src/lib.rs

```diff
@@ -139,16 +139,33 @@ pub struct PaintCtx<'a, 'b: 'a> {
     pub render_ctx: &'a mut Piet<'b>,
 }
 
-pub struct LayoutCtx {}
+pub struct LayoutCtx<'a, 'b: 'a> {
+    text: &'a mut Text<'b>,
+}
 
-pub struct EventCtx<'a> {
+/// A mutable context provided to event handling methods of widgets.
+///
+/// Widgets should call [`invalidate`] whenever an event causes a change
+/// in the widget's appearance, to schedule a repaint.
+///
+/// [`invalidate`]: #method.invalidate
+pub struct EventCtx<'a, 'b> {
+    win_ctx: &'a mut dyn WinCtx<'b>,
+    // TODO: migrate most usage of `WindowHandle` to `WinCtx` instead.
     window: &'a WindowHandle,
```

Comment:
I might just do this as part of this patch, if it isn't too much work.

Reply:
Do an invalidate? I ***really*** don't want to, it feels like widgets should have the opportunity to avoid doing extra work.

Reply:
no, I mean get rid of `WIndowHandle`.

Reply:
Ah, heh. This means that an event can't change the cursor, until I add that functionality. Separate PR maybe?

Reply:
sure, sounds reasonable.

Reply:
(I was maybe confused, it looked like there was a `WindowHandle` inside the `WinCtx`.)